adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/JaxRsImplementationGenerator.java

```diff
@@ -94,6 +93,7 @@ private TypeSpec createClassFor(final Resource resource, final Component compone
         final TypeSpec.Builder classSpecBuilder = classSpecFor(resource, component);
 
         resource.getActions().values().forEach(action -> {
+            validateActionTypeIsGetOrPost(action.getType());
```

Comment:
validateActionTypeIsSupported is a better name for this method in-case we update it later.